lib/doorkeeper/oauth/error_response.rb

```diff
@@ -32,15 +32,17 @@ def body
       end
 
       def status
-        if name == :invalid_client
+        if name == :invalid_client || name == :unauthorized_client
           :unauthorized
         else
           :bad_request
         end
       end
 
       def redirectable?
-        name != :invalid_redirect_uri && name != :invalid_client &&
+        name != :invalid_redirect_uri &&
```

Comment:
Let's introduce a const named `NON_REDIRECTABLE_STATES` or smth like this with states from above and then just:

```ruby
NON_REDIRECTABLE_STATES.include?(name)
```